### PR TITLE
Enable F401 in flake8, configure per-file disables.

### PR DESCRIPTION
**Pull request by jakkdl**
See discussion in https://github.com/python-trio/trio/pull/2666#discussion_r1240448036

### CODE REVIEW

File: trio/_core/_tests/test_run.py [hunk 1]
```diff
@@ -1598,7 +1594,7 @@ async def test_asyncio_function_inside_nursery_does_not_explode():
     # Regression test for https://github.com/python-trio/trio/issues/552
     with pytest.raises(TypeError) as excinfo:
         async with _core.open_nursery() as nursery:
-            import asyncio
+            # import asyncio
```
Comment (jakkdl) [inline trio/_core/_tests/test_run.py:1597]: I'm not sure if this `import` is to ensure `asyncio` is available or something

Comment (A5rocks) [reply trio/_core/_tests/test_run.py:1597]: I think this is just missed in PR review. cc @oremanj as author of https://github.com/python-trio/trio/pull/1178 even though that was.... 4 years ago wow.

**Comment by jakkdl** [comment]
Any objection to just merging this? It'll quickly lead to a lot of merge conflicts if I start opening type-checking PRs in parallel to it.

File: .flake8
Comment (A5rocks) [inline .flake8:1]: How about using `flake8-pyproject` to consolidate all our configuration into 1 file?

File: trio/_core/_tests/test_util.py [hunk 2]
```diff
@@ -1 +1 @@
-import pytest
+import pytest  # noqa: F401
```
Comment (A5rocks) [inline trio/_core/_tests/test_util.py:1]: Hm, why is this staying?

**Review by A5rocks — COMMENTED** [review]
Looks mostly good. I don't quite get why anything needs to have this warning ignored for (unless it's doing some runtime stuff? which kinda sucks but ig that's a different issue to fix)

File: trio/_core/_tests/test_util.py (hunk 2 above)
Comment (jakkdl) [reply trio/_core/_tests/test_util.py:1]: I assumed that importing pytest was just what the test *did* (for some strange reason), but looking at the history of the file it seems like it should probably have been removed a long time ago.

**Comment by jakkdl** [comment]
> Looks mostly good. I don't quite get why anything needs to have this warning ignored for (unless it's doing some runtime stuff? which kinda sucks but ig that's a different issue to fix)

added comments to most of the noqa's.

File: check.sh [hunk 3]
```diff
@@ -18,10 +18,8 @@ if ! black --check setup.py trio; then
     black --diff setup.py trio
 fi
 
-# Run flake8 without pycodestyle and import-related errors
-flake8 trio/ \
-    --ignore=D,E,W,F401,F403,F405,F821,F822\
-    || EXIT_STATUS=$?
+# Run flake8, configured in .flake8
```
Comment (A5rocks) [inline check.sh:21]: Forgot to update the comment!

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,18 @@
 [tool.black]
 target-version = ['py37']
 
+[tool.flake8]
+extend-ignore = ['D','E','W','F403','F405','F821','F822']
```
Comment (A5rocks) [inline pyproject.toml:5]: Eh, mind adding spacing?

**Review by A5rocks — APPROVED** [review]
Other than these 2 + fixing the formatting, this looks good!

File: pyproject.toml (hunk 4 above)
Comment (jakkdl) [reply pyproject.toml]: Ran `pyproject.toml` through https://github.com/macisamuele/language-formatters-pre-commit-hooks

**Comment by A5rocks** [comment]
Formatting CI is still failing :(

I would be a +1 on adding a nicer error message (e.g. putting a `git diff *-requirements.txt` or something) in the failure case for pip-compile.

**Comment by jakkdl** [comment]
I'm inclined to ignore the failing pypy-3.7 export symbol test since support will be dropped soon anyway.

**Comment by A5rocks** [comment]
Could you just toss an xfail at it I guess, it's required for merging :(

**Comment by A5rocks** [comment]
https://github.com/python/typing_extensions/issues/258 heh. I'll make a PR.

**Review by A5rocks — APPROVED** [review]
Rereviewing, seems good to me.

**Pull request merged**